lib/task/import/importDipObjectsTask.class.php

```diff
@@ -0,0 +1,585 @@
+      new sfCommandOption('audit', null, sfCommandOption::PARAMETER_NONE, 'Audit mode')
+    ));
+
+    $this->namespace = 'import';
+    $this->name = 'dip-objects';
+    $this->briefDescription = 'Import digital objects from Archivematica DIP using CSV file';
+
+    $this->detailedDescription = <<<EOF
+Process a CSV file to import digital objects from an Archivematica DIP to
+existing information objects in AtoM.
+
+The CSV file can be named anything, but must have the extension "csv"
+(lower-case).
+
+The CSV file must start with a header row specifying column order. A "filename"
+column must be included. Additionally, either an "identifier" or a "slug"
+column must be included (not both: one of these will be used to specify
+information objects to import to).
+
+The import script expects that Archivematica will have modified the filename
+so:
+
+a) The original object file has been converted to a derivative with a
+   correspondiing extension ("jpg" for example)
+
+b) A UUID has been pre-pended to the filename (for example:
+   "815da5cf-f49f-41f5-aa5d-c40d9d4dec3c-MARBLES.jpg")
+
+Note: Filenames must be unique (without UUID) to avoid colliding on import.
+
+The undo-log-dir option can be used to log which information objects have
+digital objects added to them. This log can be used, in event of an incomplete
+import, to either establish where the import stopped or to manually remove the
+imported digital objects. Undo logs contain two columns: the ID of the
+information object to which objects have beem imported and the DIP directory
+the objects were imported from.
+
+The audit option can be used to verify that all objects specified in a DIP's
+CSV file were imported. If any are found to be missing the object filename
+will be output.
+EOF;
+  }
+
+  /**
+   * @see sfTask
+   */
+  public function execute($arguments = array(), $options = array())
+  {
+    $this->checkArgumentsAndOptions($arguments, $options);
+
+    sfContext::createInstance($this->configuration);
+
+    QubitSearch::getInstance()->disabled = false;
+
+    $databaseManager = new sfDatabaseManager($this->configuration);
+    $this->conn = $databaseManager->getDatabase('propel')->getConnection();
+
+    sfConfig::set('app_upload_dir', $this->getUploadDir());
+
+    // Set dip directory
+    $this->dipDir = $arguments['dip'];
+
+    // Set undo log filename, if undo log directory is specified
+    $undoLog = null;
+    if (isset($options['undo-log-dir']))
+    {
+      $undoLog = rtrim($options['undo-log-dir'], '/') .'/'. date('Y-m-d') .'-'. basename($this->dipDir);
+    }
+
+    // Set path to DIP objects
+    $objectsPath = rtrim($this->dipDir, '/') .'/objects';
+    $this->logSection(sprintf("Looking for objects in: %s", $this->dipDir));
+
+    // Parse CSV file and import/audit objects
+    $digitalObjects = $this->parseCsvData($this->openFirstCsvFile(), $objectsPath);
```

Comment:
You could sent the '$objectsPath' to the 'openFirstCsvFile()' function, as this variable already has the possible extra slash trimed, and in that function we're not doing the same.